ml_genn/model.py

```diff
@@ -356,45 +338,30 @@ def convert_tf_model(tf_model, converter=Simple(),
                              if n in tf_model_nodes]
             tf_out_layers_all[tf_layer] = tf_out_layers
 
-        # function for traversing upstream layers
-        def traverse_tf_in_layers(in_layers):
-            new_in_layers = set(in_layers)
-            final_in_layers = set()
-
-            while new_in_layers:
-                layer = new_in_layers.pop()
-                if isinstance(layer, ignored_tf_layers):
-                    new_in_layers.update(tf_in_layers_all[layer])
-                else:
-                    final_in_layers.add(layer)
-
-            return final_in_layers
-
-        # function for traversing downstream layers
-        def traverse_tf_out_layers(out_layers):
-            new_out_layers = set(out_layers)
-            final_out_layers = set()
-
-            while new_out_layers:
-                layer = new_out_layers.pop()
-                if isinstance(layer, ignored_tf_layers):
-                    new_out_layers.update(tf_out_layers_all[layer])
-                else:
-                    final_out_layers.add(layer)
-
-            return final_out_layers
-
 
         # Perform any pre-compilation tasks
         pre_compile_output = converter.pre_compile(tf_model)
 
-        # configure ML GeNN model build process
-        mlg_model_inputs = []
-        mlg_model_outputs = []
-        mlg_layer_lookup = {}
+        # configure model build process
+        class LayerConfig(object):
+            def __init__(self):
```

Comment:
why not have constructor arguments for ``is_input``, ``has_activation``, ``name``, ``shape`` and ``neurons`` (and defaults if any are ever optional) so you can go e.g.:
```python
config = LayerConfig(is_input=True, has_activation=True, name=tf_layer.name, shape=tf_layer.output_shape[0][1:], 
                     neurons=converter.create_input_neurons(pre_compile_output))
```